### PR TITLE
chore(icon): remove classic theme - FE-3112

### PR DESCRIPTION
**Pull request by mkrds**
### Proposed behaviour
This PR removes the classic theme support from Icon component

### Checklist

- [x] Commits follow our style guide
- [x] Screenshots are included in the PR if useful
- [x] All themes are supported if required
- [x] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [x] Storybook added or updated if required
- [x] Typescript `d.ts` file added or updated if required
- [x] Carbon implementation and Design System documentation are congruent

### CODE REVIEW

**Review by DlgSHi — COMMENTED** [review]
I don't think we need to create new examples with `icon` in `chromatic` because we have already covered all `variants` of `icon` component. Probably we could `disable` all `chromatic` tests for `mdx` file. And move `all` back from `Test` directory.

**Review by sjohnson-sage — DISMISSED** [review]

**Comment by DipperTheDan** [comment]
Noticed a little bug on the `With tooltip` example when on the Canvas tab. The tooltip doesn't display however I think it's trying to render above the icon.

icon_tooltip_bug

File: cypress/support/step-definitions/themes-steps.js [hunk 1]
```diff
@@ -25,7 +25,7 @@ When('I open {string} component with theme {string} knobs story', (componentName
 });
 
 When('I open Icon component with theme {string}', (themeName) => {
-  visitComponentUrlByTheme('icon', themeName, BUSINESS_BGTHEME);
+  visitComponentUrlByTheme('icon-test', themeName, BUSINESS_BGTHEME);
```
Comment (DlgSHi) [inline cypress/support/step-definitions/themes-steps.js:28]: This step is already deleted on the `master` branch.

**Review by DlgSHi — COMMENTED** [review]
please update your major branch with `master` and apply new `approach` to open `Icon` component and check theming colours.

**Review by sjohnson-sage — DISMISSED** [review]

**Review by ianforster-sage — DISMISSED** [review]

**Review by DlgSHi — DISMISSED** [review]

**Review by ianforster-sage — DISMISSED** [review]

**Review by sjohnson-sage — DISMISSED** [review]

**Review by DlgSHi — APPROVED** [review]

**Review by ianforster-sage — APPROVED** [review]

**Review by sjohnson-sage — APPROVED** [review]

**Review by DlgSHi — APPROVED** [review]

**Review by sjohnson-sage — APPROVED** [review]

**Review by ianforster-sage — APPROVED** [review]

**Pull request merged**